lib/adagrams.rb

```diff
@@ -0,0 +1,151 @@
+letter_quantities = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+}
+$letter_pool = []
+letter_quantities.each do |letter, quantity|
+  quantity.times do
+    $letter_pool << letter.to_s
+  end
+end
+
+$letter_scores = {
+  1 => ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+  2 => ["D", "G"],
+  3 => ["B", "C", "M", "P"],
+  4 => ["F", "H", "V", "W", "Y"],
+  5 => ["K"],
+  8 => ["J", "X"],
+  10 => ["Q", "Z"],
+}
+
+# Wave 1
+def draw_letters
+  letters_in_hand = $letter_pool.sample(10)
+  return letters_in_hand
+end
+
+# Wave 2
+def uses_available_letters?(input, letters_in_hand)
+  input = input.upcase.split("")
+  # Make a copy of the input array and keep the original 
+  # array unchanged. The original input needs to remain in tact so
+  # that Ruby can continue to iterate through it the 
+  # uses_available_letters methods.
```

Comment:
This is a great use of a comment: to add detail about something that feels a little weird and needs explanation!